app/src/main/java/com/framgia/soundclound/screen/playtrack/PlayTrackViewModel.java

```diff
@@ -60,4 +161,48 @@ public void onClickDownload(View view) {
 
     }
 
+    @Override
+    public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
+
+    }
+
+    @Override
+    public void onStartTrackingTouch(SeekBar seekBar) {
+//         asyncTask.cancel(true);
+    }
+
+    @Override
+    public void onStopTrackingTouch(SeekBar seekBar) {
+        int currentPosition = StringUtil.progressToTimer(
+                seekBar.getProgress(), mMusicService.getDuration());
+        mMusicService.seekto(currentPosition);
+    }
+
+    // asyntask
+    public class TimeAsysnTask extends AsyncTask<Void, Void, Void> {
```

Comment:
WARNING: [StaticFieldLeak] This AsyncTask class should be static or leaks might occur (com.framgia.soundclound.screen.playtrack.PlayTrackViewModel.TimeAsysnTask)
 A static field will leak contexts.

Non-static inner classes have an implicit reference to their outer class. If that outer class is for example a `Fragment` or `Activity`, then this reference means that the long-running handler/loader/task will hold a reference to the activity which prevents it from getting garbage collected.

Similarly, direct field references to activities and fragments from these longer running instances can cause leaks.

ViewModel classes should never point to Views or non-application Contexts.